streams/src/main/java/org/apache/kafka/streams/processor/UsePreviousTimeOnInvalidTimestamp.java

```diff
@@ -37,11 +37,14 @@
  * using this extractor effectively provides <i>ingestion-time</i> semantics.
  * <p>
  * If you need <i>processing-time</i> semantics, use {@link WallclockTimestampExtractor}.
+ * @deprecated UsePreviousTimeOnInvalidTimestamp will be deprecated in the future, please use
```

Comment:
We usually only include the version (as the JavaDocs will stay the same if future version). Including the name of the class is redundant, and it's also deprecated as of _now_ (not in the future) -- we don't need to explain what deprecation means IMHO.

I would suggest:
```
@deprecated since 2.5. Use {@link UsePartitionTimeOnInvalidTimestamp} instead.
```